src/SourceLocator/Ast/Locator.php

```diff
@@ -27,6 +27,11 @@ class Locator
      */
     private $parser;
 
+    /**
+     * @var array|string[]
+     */
+    private static $cache = [];
```

Comment:
Move this to a `CachedLocator`. Adding a cache, and ESPECIALLY a static one is a no-go here.